mmengine/hooks/early_stopping_hook.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import warnings
+from typing import List, Optional, Union
+
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+DATA_BATCH = Optional[Union[dict, tuple, list]]
+
+
+@HOOKS.register_module()
+class EarlyStoppingHook(Hook):
+    """Early stop the training when the metric reached a plateau.
+
+    Args:
+        metric (str): The metric key to decide early stopping.
+        rule (str, optional): Comparison rule. Options are 'greater',
+            'less'. Defaults to None.
+        delta(float, optional): Minimum difference to continue the training.
+            Defaults to 0.01.
+        pool_size (int, optional): The number of experiments to consider.
+            Defaults to 5.
+        patience (int, optional): Maximum number of tolerance.
```

Comment:
Ahhh, it could be better to describe the patience in more detailed

Reply:
I borrowed the description from https://docs.ray.io/en/latest/_modules/ray/tune/stopper/experiment_plateau.html#ExperimentPlateauStopper.

Is the description from https://keras.io/api/callbacks/early_stopping/ better?

Reply:
Yep, I think the description of https://keras.io/api/callbacks/early_stopping/ is better. BTW, the `epoch` in the description is not accurate, will `the times of validation` be better?